shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.h

```diff
@@ -61,6 +61,7 @@ void ResetAnchor(CALayer* layer);
 
 class IOSContextGL;
 class IOSSurface;
+class IOSSurfaceFactory;
```

Comment:
Nit: You are suppose to avoid forward declarations in C++

https://google.github.io/styleguide/cppguide.html#Forward_Declarations

Reply:
There is a cyclical dependency here. `IOSSurfaceFactory` holds a platform views controller via `SetPlatformViewsController` and platform views controller also needs a surface factory for creating surfaces. I will simplify this in the follow-up PRs.